### PR TITLE
Update coq dev opam packages; add CoqIDE

### LINKED ISSUE #202: Coq is not uninstalled completely
When doing `opam uninstall coq` (I had version 8.7.dev installed), there are some files left in the opam directories:
```
./man/man1/coqdoc.1
./man/man1/coqdep.1
./man/man1/coqwc.1
./man/man1/coqtop.1
./man/man1/coqmktop.1
./man/man1/coq_makefile.1
./man/man1/coqc.1
./man/man1/coq-tex.1
./man/man1/coqtop.opt.1
./man/man1/coqide.1
./man/man1/coqchk.1
./man/man1/coqtop.byte.1
./bin/coqtop.byte
./share/texmf/tex/latex/misc/coqdoc.sty
./share/emacs/site-lisp/coqdoc.sty
./share/emacs/site-lisp/coq-font-lock.el
./share/emacs/site-lisp/coq-inferior.el
```

### PR DESCRIPTION
**Pull request by RalfJung**
This pulls in some changes from the "official" opam package in the main repository. I will also submit a PR there, mostly syncing the two.

Fixes #202

### CODE REVIEW

File: core-dev/packages/coq.dev/opam [hunk 1]
```diff
@@ -1,23 +1,55 @@
...
+  [make "install-byte"]
+]
+remove: [
+  ["rm" "-R" "%{lib}%/coq" "%{share}%/coq"]
+  ["rm"
+  "%{man}%/man1/coqc.1"
+  "%{man}%/man1/coqchk.1"
+  "%{man}%/man1/coqdep.1"
+  "%{man}%/man1/coqdoc.1"
+  "%{man}%/man1/coqide.1"
```
Comment (herbelin) [inline core-dev/packages/coq.dev/opam]: I'm lost. If there is `-coqide no`, why to remove `coqide.1`?

Comment (herbelin) [inline core-dev/packages/coq.dev/opam]: If there is -coqide no, why to remove `coqide.1`? Isn't there a "bug" in the install target?

Comment (RalfJung) [reply core-dev/packages/coq.dev/opam]: Hm, good question. I think that indeed the `coq` opam package does install this manpage, let me check again.

Comment (RalfJung) [reply core-dev/packages/coq.dev/opam]: Indeed, this is the case. So for now, the opam package is correct -- the `uninstall` should un-do the `install`, after all.

I will open an issue about the `coqide` man page.

**Comment by RalfJung** [comment]
The PR on the "default" repository side is at https://github.com/ocaml/opam-repository/pull/10987.

**Pull request merged**